src/test/java/com/kodilla/ecommercee/domain/GroupRepositoryTestSuite.java

```diff
@@ -0,0 +1,145 @@
+package com.kodilla.ecommercee.domain;
+
+import com.kodilla.ecommercee.repository.GroupRepository;
+import com.kodilla.ecommercee.repository.ProductRepository;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class GroupRepositoryTestSuite {
+    @Autowired
+    private ProductRepository productRepository;
+
+    @Autowired
+    private GroupRepository groupRepository;
+
+    private Group group1;
+    private Group group2;
+    private Product product1;
+    private Product product2;
+    private Product product3;
+    private Product product4;
+
+    @Test
+    public void testGroupRepositorySave() {
+        //Given
+        createData();
+
+        //When
+        groupRepository.save(group1);
+
+        //Then
+        assertNotEquals(0, group1.getId());
+
+        //CleanUp
+        deleteData();
+    }
+
+    @Test
+    public void testGroupRepositoryDelete() {
+        //Given
+        createData();
+        groupRepository.save(group1);
+
+        //When
+        groupRepository.deleteById(group1.getId());
+
+
+        //Then
+        assertEquals(Optional.empty(), groupRepository.findById(group1.getId()));
+
+    }
+
+    @Test
+    public void testGroupRepositoryUpdate() {
+        //Given
+        createData();
+
+        //When
+        groupRepository.save(group1);
```

Comment:
ta linia do przeniesienia do Given

W Given ma być stan zastany, w When to co testujesz wraz z ewentualnymi dodatkowymi operacjami towarzyszącymi - tak jak tutaj stworzenie Stringa, zaktualizowanie obiektu Encji, a dopiero aktualizacja w bazie, a w Then sprawdzasz założenia